src/ui-components/empty-state/empty-state.component.tsx

```diff
@@ -0,0 +1,41 @@
+import React from "react";
+import SummaryCard from "../cards/summary-card.component";
+import styles from "./empty-state.css";
+import { match } from "react-router-dom";
+
+export default function EmptyState(props: EmptyStateProps) {
+  return (
+    <SummaryCard
+      name={props.name}
+      addComponent={props.addComponent}
+      showComponent={props.showComponent}
+    >
+      <div
+        style={props.styles}
+        className={`omrs-medium ${styles.emptyStateText}`}
+      >
+        <p className="omrs-type-body-regular">{props.displayText}</p>
+      </div>
+    </SummaryCard>
+  );
+}
+
+type EmptyStateProps = {
+  name: string;
+  displayText: string;
+  styles?: React.CSSProperties;
+  addComponent?: React.FC<TProps | DataCaptureComponentProps>;
+  showComponent?: () => void;
+};
+
+type TProps = {
```

Comment:
When I see `TProps` I always assume its a generic type. Can we give this a more accurate name, e.g., `RouteBaseComponentProps`?

Reply:
Sure. That's a great variable name.